src/backend/dx12/src/command.rs

```diff
@@ -250,9 +351,43 @@ impl command::RawCommandBuffer<Backend> for CommandBuffer {
         &mut self,
         layout: &n::PipelineLayout,
         first_set: usize,
-        sets: &[&()],
+        sets: &[&n::DescriptorSet],
     ) {
-        unimplemented!()
+        unsafe {
+            self.raw.SetGraphicsRootSignature(layout.raw);
+
+            // Bind descriptor heaps
+            // TODO: Can we bind them always or only once?
+            //       Resize while recording?
+            let mut heaps = [
+                sets[0].heap_srv_cbv_uav.as_mut() as *mut _,
+                sets[0].heap_samplers.as_mut() as *mut _
+            ];
+            self.raw.SetDescriptorHeaps(2, heaps.as_mut_ptr())
+        }
+
+        let mut table_id = 0;
+        for (i, table) in layout.tables.iter().enumerate() {
```

Comment:
let's split this into 2 loops, one `for table in &layout.tables[..first_set]`, and another `for (set, _) in sets.iter_mut().zip(layout.tables[first_set..].iter())`